packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/categories/ExpoNotificationCategoriesModule.java

```diff
@@ -0,0 +1,163 @@
+package expo.modules.notifications.notifications.categories;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.os.ResultReceiver;
+
+import org.unimodules.core.ExportedModule;
+import org.unimodules.core.Promise;
+import org.unimodules.core.interfaces.ExpoMethod;
+import org.unimodules.core.errors.InvalidArgumentException;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+
+import expo.modules.notifications.notifications.NotificationSerializer;
+import expo.modules.notifications.notifications.service.BaseNotificationsService;
+import expo.modules.notifications.notifications.model.NotificationCategory;
+import expo.modules.notifications.notifications.model.NotificationAction;
+import expo.modules.notifications.notifications.model.TextInputNotificationAction;
+
+public class ExpoNotificationCategoriesModule extends ExportedModule {
+  private static final String EXPORTED_NAME = "ExpoNotificationCategoriesModule";
+  private static final String IDENTIFER = "identifier";
+  private static final String BUTTON_TITLE = "buttonTitle";
+  private static final String OPTIONS = "options";
+  private static final String OPENS_APP_TO_FOREGROUND = "opensAppToForeground";
+  private static final String TEXT_INPUT_OPTIONS = "textInput";
+  private static final String SUBMIT_BUTTON_TITLE = "submitButtonTitle";
+  private static final String PLACEHOLDER = "placeholder";
+
+  public ExpoNotificationCategoriesModule(Context context) {
+    super(context);
+  }
+
+  @Override
+  public String getName() {
+    return EXPORTED_NAME;
+  }
+
+  @ExpoMethod
+  public void getNotificationCategoriesAsync(final Promise promise) {
+    BaseNotificationsService.enqueGetCategories(getContext(), new ResultReceiver(null) {
+      @Override
+      protected void onReceiveResult(int resultCode, Bundle resultData) {
+        super.onReceiveResult(resultCode, resultData);
+        Collection<NotificationCategory> categories = resultData.getParcelableArrayList(BaseNotificationsService.CATEGORIES_KEY);
+        if (resultCode == BaseNotificationsService.SUCCESS_CODE && categories != null) {
+          promise.resolve(serializeCategories(categories));
+        } else {
+          Exception e = resultData.getParcelable(BaseNotificationsService.EXCEPTION_KEY);
+          promise.reject("ERR_CATEGORIES_FETCH_FAILED", "A list of notification categories could not be fetched.", e);
+        }
+      }
+    });
+  }
+
+  @ExpoMethod
+  public void setNotificationCategoryAsync(final String identifier, List<HashMap<String, Object>> actionArguments, final Promise promise) {
+    List <NotificationAction> actions = new ArrayList();
+    for (HashMap<String, Object> actionMap : actionArguments) {
+      HashMap<String, Object> actionOptions = getActionOptions(actionMap);
```

Comment:
this `getting` code could become easier if we used `MapArguments` class which may come in handy here :)

Reply:
what a convenient class 😉